app/main.py

```diff
@@ -1,2 +1,67 @@
+from typing import Any
+from collections.abc import Hashable
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.__len = 8
+        self.dictionary = [None] * self.__len
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        if "__hash__" not in self.__dir__():
```

Comment:
The check for hashability is incorrect. Instead of checking '__hash__' in self.__dir__(), you should check if the key is an instance of Hashable using isinstance(key, Hashable).